docs/DeveloperGuide.md

```diff
@@ -1,37 +1,300 @@
+- Receives user's input and splits it into the Command Type and Description using the `Functions` class.
+- Uses the `itemList`, which is an `ExpenditureList` object, to instantiate a `Command` object based on the Command Type.
+- Returns the `Command` object that can then be executed.
+
+We pass in the `itemList` to the `Command` object instead of using a global variable to ease testing. This way, we can 
+add, delete and update `Expenditure` entries in a new `itemList` during testing without affecting the actual`itemList`.
+
+### Command component 
+The source can be found in [`command`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)  
+Fig 5 - Command Class Diagram
+
+The Command component consists of Command abstract class, `AddCommand`, `ByeCommand`, `DeleteCommand`, `HelpCommand`
+, `ListCommand`and `UpdateCommand` that extends the `Command` class.
+
+The Command component:
+- Provides all the Command classes which can be instantiated by `Parser.parseCommand()`. The Command objects can then be 
+executed. Only 1 Command object can be created.
+- Includes a `Command.executeCommand()` method in each Command classes which performs the relevant command and throws 
+exceptions if an error is encountered. The error is then handled.
+
+### Storage component 
+The source can be found in [`Storage.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Storage.java)
+
+![storage_diagram](images/storage_diagram.png)  
+Fig 6 - Storage Class Diagram
+
+The Storage component consists of `Storage` class.
+
+The Storage component:
+- `MMM` class initialises a `Storage` object upon start up. The `Storage` class consists of 
+`Storage.load()` and `Storage.save()` methods. 
+- Concurrently, `MMM` will call the `Storage.load()` method and load any data that is stored on the hard disk.
+- `MMM` calls the `Storage.save()` method and stores remaining data onto the hard disk when the program exits.
+
+
+## Implementation
+This section describes some noteworthy details on how certain features of MindMyMoney are implemented.
+
+### AddCommand Feature
+The source code can be found in [`AddCommand.java`](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command/AddCommand.java)
+
+The AddCommand feature allows users to add in an expenditure, a new credit card or their income using a single command. 
+This component provides speed and ease of use by only requiring a single line of input.
+
+#### AddCommand Subcomponents
+The AddCommand component consists of 3 subcomponents. These features are differentiated by their flags.
+- Add expenditure component.
+- Add credit card component.
+- Add income component.
+
+The sequence diagram below shows the interactions of when an `AddCommand` is parsed.  
+
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
```

Comment:
<img width="412" alt="Screen Shot 2022-04-01 at 10 22 27 AM" src="https://user-images.githubusercontent.com/55422893/161182071-b2bf652a-57da-4189-aa92-dc4f16c5629a.png">
Shouldn't the return arrow be outside of the alt frame?